### DISCUSSION TITLE
Client connection scalability

### DISCUSSION THREAD

**Discussion by dbi1** [postgresml/pgcat#653] [Q&A]
Hi,

pgcat looks really good from everything I read about it.

I am currently trying to decide between pgcat and Supavisor. Do you think pgcat would scale as well (or maybe better) than Supavisor, when it comes to # of client connections? I was thinking Rust's 'fearless concurrency' might play well here?

Thank you for your thoughts!

**Comment by levkk** [comment]
PgCat is only limited by what Tokio, and inherently the operating system/standard library epoll implementation can handle. I see no reason why you can't scale PgCat to 1M+ clients/servers. Remember to increase your open file description limits [1], since that's the only limit we can hit.

[1]: https://unix.stackexchange.com/questions/8945/how-can-i-increase-open-files-limit-for-all-processes

**Reply by dbi1** [reply]
Very, very cool, thank you!

Btw, might there be any plans to add primary failover at some point?
